src/timeseriesflattener/flattened_dataset.py

```diff
@@ -526,196 +615,174 @@ def _add_incident_outcome(
-    ) -> DataFrame:
-        """Ensure all prediction times are represented in the returned
+                self.unprocessed_specs.outcome_specs.remove(spec)
 
-        dataframe.
+        temporal_batch = self.unprocessed_specs.outcome_specs
+        temporal_batch += self.unprocessed_specs.predictor_specs
 
-        Args:
-            df (DataFrame): Dataframe with prediction times but without uuid.
-            pred_times_with_uuid (DataFrame): Dataframe with prediction times and uuid.
-            pred_time_uuid_colname (str): Name of uuid column in both df and pred_times_with_uuid.
+        if self.drop_pred_times_with_insufficient_look_distance:
+            self._drop_pred_time_if_insufficient_look_distance()
 
-        Returns:
-            DataFrame: A merged dataframe with all prediction times.
-        """
-        return pd.merge(
-            pred_times_with_uuid,
-            df,
-            how="left",
-            on=pred_time_uuid_colname,
-            suffixes=("", "_temp"),
-        ).drop(["timestamp_pred"], axis=1)
+        if len(temporal_batch) > 0:
+            self._add_temporal_batch(temporal_batch=temporal_batch)
 
-    @staticmethod
-    def _resolve_multiple_values_within_interval_days(
-        resolve_multiple: Callable,
-        df: DataFrame,
-        pred_time_uuid_colname: str,
-    ) -> DataFrame:
-        """Apply the resolve_multiple function to prediction_times where there
+    def _check_that_spec_df_has_required_columns(self, spec: AnySpec):
+        """Check that df has required columns."""
+        # Find all attributes in self that contain col_name
+        required_columns = [self.id_col_name]
 
-        are multiple values within the interval_days lookahead.
+        if not isinstance(spec, StaticSpec):
+            required_columns += [self.timestamp_col_name]
 
-        Args:
-            resolve_multiple (Callable): Takes a grouped df and collapses each group to one record (e.g. sum, count etc.).
-            df (DataFrame): Source dataframe with all prediction time x val combinations.
-            pred_time_uuid_colname (str): Name of uuid column in df.
+        for col in required_columns:
+            if col not in spec.values_df.columns:  # type: ignore
+                raise ValueError(f"Missing required column: {col}")
 
-        Returns:
-            DataFrame: DataFrame with one row pr. prediction time.
-        """
-        # Convert timestamp val to numeric that can be used for resolve_multiple functions
-        # Numeric value amounts to days passed since 1/1/1970
-        try:
-            df["timestamp_val"] = (
-                df["timestamp_val"] - dt.datetime(1970, 1, 1)
-            ).dt.total_seconds() / 86400
-        except TypeError:
-            log.info("All values are NaT, returning empty dataframe")
+    def add_spec(
+        self,
+        spec: Union[list[AnySpec], AnySpec],
+    ):
+        """Add a specification to the flattened dataset."""
```

Comment:
As this is the one of the main user facing functions it should have more thorough documentation. Ideally a minimal example as well